### PR TITLE
Add option to fetch RSA primitives

### PR DESCRIPTION
**Pull request by ptoffy**

### CODE REVIEW

File: Sources/JWTKit/RSA/SubjectPublicKeyInfo.swift [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+extension RFC5480AlgorithmIdentifier {
+    static let rsa256 = RFC5480AlgorithmIdentifier(
+        algorithm: .AlgorithmIdentifier.sha256WithRSAEncryption
+    )
+
+    static let rsa384 = RFC5480AlgorithmIdentifier(
+        algorithm: .AlgorithmIdentifier.sha384WithRSAEncryption
+    )
+
+    static let rsa521 = RFC5480AlgorithmIdentifier(
```
Comment (MahdiBM) [inline Sources/JWTKit/RSA/SubjectPublicKeyInfo.swift:103]: 521 😱

Comment (gwynne) [reply Sources/JWTKit/RSA/SubjectPublicKeyInfo.swift:103]: Not a typo.

Comment (MahdiBM) [reply Sources/JWTKit/RSA/SubjectPublicKeyInfo.swift:103]: What the heck 😶

Comment (gwynne) [reply Sources/JWTKit/RSA/SubjectPublicKeyInfo.swift:103]: A somewhat bizarre mathematical quirk of the NIST curves that were chosen. See https://crypto.stackexchange.com/questions/6219/why-do-the-elliptic-curves-recommended-by-nist-use-521-bits-rather-than-512 for some gory details.

Comment (ptoffy) [reply Sources/JWTKit/RSA/SubjectPublicKeyInfo.swift:103]: Actually that _is_ a typo, RSA is not based on elliptic curves. It should be 512 as that's the hashing algorithm

Comment (gwynne) [reply Sources/JWTKit/RSA/SubjectPublicKeyInfo.swift:103]: Damn, you're right. Nice catch! (And doubly so with me incorrectly arguing the opposite 😅)

File: Tests/JWTKitTests/RSATests.swift [hunk 2]
```diff
@@ -73,6 +73,57 @@ final class RSATests: XCTestCase {
...
+        XCTAssertEqual(exponent, publicExponent)
+    }
+
+    func testGetPrivateKeyPrimitivesWhenKeyIsPublicThrows() async throws {
+        let publicKey = try RSAKey(modulus: modulus, exponent: publicExponent)
+        XCTAssertThrowsError(try publicKey.getPrivateKeyPrimitives())
+    }
+
+    func testGetPrivateKeyPrimitivesFromNonRawBuiltKey() async throws {
+        let privateKey = try RSAKey.private(pem: """
```
Comment (gwynne) [inline Tests/JWTKitTests/RSATests.swift:104]: Can we also test that `BEGIN RSA PRIVATE KEY` will not decode as a PKCS#8 `PrivateKeyInfo` (or the public ones if that's easier) even if the data has the correct structure? (Conforming implementations are supposed to reject anything other than `BEGIN/END PRIVATE/PUBLIC KEY` for PKCS#8 PEM)

**Review by gwynne — COMMENTED** [review]
Just one nit, looks pretty good! As for attribution to the test code copied from swift-asn1, update the `NOTICES.txt` file accordingly.

**Pull request merged**